File: MainActivity.java
package com.example.appnumbershapes;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    class Number {
        int number;

        public Number() {}
        public Number(int n) {
            number = n;
        }
        public boolean isSquare() {
            double squareRoot = Math.sqrt(number);
            if(squareRoot == Math.floor(squareRoot)) {
                return true;
            }
            return false;
        }
        public boolean isTriangular() {
            int x=1;
            int triangularNum=1;

            while(triangularNum<number) {
                x++;
                triangularNum += x;
            }
            if(triangularNum == number ) {
                return true;
            }
            return false;
        }


    }
    public void testNumber(View view){
        Log.i("Info","Button Pressed.");
        EditText editText = (EditText) findViewById(R.id.editTextNumber);
        String message;
        if (editText.getText().toString().isEmpty()) {
            message = "Please enter a number.";
        } else {
            Number myNumber = new Number();
            myNumber.number = Integer.parseInt(editText.getText().toString());
            message = editText.getText().toString();
            if (myNumber.isSquare() && myNumber.isTriangular()) {
                message += " is both square and triangular.";
            } else if (myNumber.isSquare()) {
                message += " is square but not triangular.";
            } else if (myNumber.isTriangular()) {
                message += " is triangular but not square.";
            } else {
                message += " is neither square nor triangular.";
            }
        }
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();

    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: MainActivity.kt
package com.example.snapchatclone

import android.content.ContentValues.TAG
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ktx.database
import com.google.firebase.ktx.Firebase

class MainActivity : AppCompatActivity() {

    var emailEditText: EditText? = null
    var passwordEditText: EditText? = null
    val mAuth = FirebaseAuth.getInstance()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        emailEditText = findViewById(R.id.emailEditText)
        passwordEditText = findViewById(R.id.passwordEditText)


        if(mAuth.currentUser != null) {
            login()
        }

    }

    fun goClicked(view: View) {
        //Check if we can log in the user
        mAuth.signInWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString())
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    // Sign in success, update UI with the signed-in user's information
                    login()
                } else {
                    //if it doesn't work -> sign up
                    mAuth.createUserWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString()).addOnCompleteListener(this) { task ->
                        if (task.isSuccessful) {
                            //FirebaseDatabase.getInstance().getReference().child("users").child("email").push().setValue("123").
                            FirebaseDatabase.getInstance().reference.child("users").child(task.result?.user!!.uid).child("email").setValue(emailEditText?.text.toString()).addOnSuccessListener {
                                Toast.makeText(this,"SUCCESS", Toast.LENGTH_SHORT).show()

                            }.addOnFailureListener{
                                Toast.makeText(this,"FAIL",Toast.LENGTH_SHORT).show()
                            }
                            login()
                        } else {
                            Toast.makeText(this,"Login, failed. Try Again.", Toast.LENGTH_SHORT).show()
                        }
                    }
                }
            }


    }

    fun login() {
        //move to next activity
        val intent = Intent(this, SnapsActivity:: class.java)
        startActivity(intent)
    }
}